### PR TITLE
[ONNX CI] Optionally rebuild nGraph during test

### PR DESCRIPTION
**Pull request by mchrusci**
CI will rebuild nGraph during test as long as there is `REBUILD NGRAPH` string in the last comment in PR.

### CODE REVIEW

**Comment by mchrusci** [comment]
TEST NOW REBUILD NGRAPH

**Comment by mchrusci** [comment]
TEST NOW REBUILD NGRAPH

**Comment by mchrusci** [comment]
TEST NOW

**Comment by mchrusci** [comment]
TEST NOW

**Comment by mchrusci** [comment]
CLEAR CACHE

**Comment by mchrusci** [comment]
CLEAR CACHE

**Comment by mchrusci** [comment]
CLEAR CACHE

**Comment by mchrusci** [comment]
CLEAR CACHE

**Comment by mchrusci** [comment]
BUILD NOW

**Comment by mchrusci** [comment]
TEST NOW

File: .ci/jenkins/Jenkinsfile [hunk 1]
```diff
@@ -65,10 +67,13 @@ def RunDockerContainers(configurationMaps) {
...
         UTILS.PropagateStatus("Run_docker_containers", configMap["dockerContainerName"])
+        if ( CI_PARAMS.contains("REBUILD NGRAPH") ) {
+            NGRAPH_DIRECTORY="/home"
+            ENVPREP_ARGS="--rebuild-ngraph"
+        }
         sh """
-            docker cp ${CI_ROOT}/prepare_environment.sh ${configMap["dockerContainerName"]}:/home
             docker cp ${CI_ROOT}/utils/docker.sh ${configMap["dockerContainerName"]}:/home
-            docker exec ${configMap["dockerContainerName"]} ./home/prepare_environment.sh
+            docker exec ${configMap["dockerContainerName"]} bash -c "/root/${CI_ROOT}/prepare_environment.sh ${ENVPREP_ARGS}"
```
Comment (arogowie-intel) [inline .ci/jenkins/Jenkinsfile]: I see that tests are passing but I'm concerned whether this path is correct: `/root/${CI_ROOT}/prepare_environment.sh`  -> which is evaluated (as far as I understand it) into `/root/.ci/jenkins/prepare_environment.sh`. Shouldn't this contain somewhere inside `ngraph` directory ?

File: .ci/jenkins/prepare_environment.sh [hunk 2]
```diff
@@ -16,39 +16,67 @@
...
+    if [ -e ./ngraph ]; then
+        cd ./ngraph
+        # If ngraph repo is up to date, and wheel exist - no need to rebuild it so exit
+        if [[ $(git pull) == *"Already up-to-date"* && -n $(find /home/ngraph/python/dist/ -name 'ngraph*.whl') ]]; then
+            exit 0
+        fi
+    else
+        git clone https://github.com/NervanaSystems/ngraph.git
+    fi
+    build_ngraph "/home"
```
Comment (arogowie-intel) [inline .ci/jenkins/prepare_environment.sh]: Why you build it once in `"/root"` and the other time in `"/home"`?

File: .ci/jenkins/Jenkinsfile (hunk 1 above)
Comment (mchrusci) [reply .ci/jenkins/Jenkinsfile]: Yes it should and in fact it does. This is path inside docker container, where root is mapped to directory containing cloned `ngraph-onnx` repo.

File: .ci/jenkins/prepare_environment.sh (hunk 2 above)
Comment (mchrusci) [reply .ci/jenkins/prepare_environment.sh]: Binaries build in `/home` are meant to be cached and reused by later builds. `/root' is path in docker container mapped to Jenkins job workspace, which gets cleared after the job completes.

Comment (mchrusci) [reply .ci/jenkins/prepare_environment.sh]: Building nGraph in `/root` is case for building nGraph from scratch.

**Comment by tsocha** [comment]
Maybe we should add this two flags to ngraph: `-DNGRAPH_TOOLS_ENABLE=FALSE` `-DNGRAPH_UNIT_TEST_ENABLE=FALSE`

File: .ci/jenkins/prepare_environment.sh (hunk 2 above)
Comment (arogowie-intel) [reply .ci/jenkins/prepare_environment.sh]: Great, thanks!

**Pull request merged**